src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -385,7 +385,12 @@ private FunctionExpression ConvertObject(ObjectSyntax syntax)
             int index = 0;
             foreach (var propertySyntax in syntax.Properties)
             {
-                parameters[index] = new JTokenExpression(propertySyntax.TryGetKeyText());
+                parameters[index] = propertySyntax.Key switch
+                {
+                    IdentifierSyntax identifier => new JTokenExpression(identifier.IdentifierName),
+                    StringSyntax @string => ConvertString(@string),
+                    _ => throw new NotImplementedException($"Encountered an unexpected type '{propertySyntax.Key.GetType().Name}' when generating object's property name.")
```

Comment:
Please could you add a test for the file supplied in the issue to ScenarioTests.cs? Just to make sure we don't accidentally regress this in the future.

Reply:
I added it to variables.bicep integration tests, but will add it there as well.

Reply:
Thanks!

Reply:
Added!